conductor/migrations/20240321181458_create_deleted_instances_table.up.sql

```diff
@@ -0,0 +1,8 @@
+-- Up migration
+CREATE TABLE deleted_instances (
+    namespace VARCHAR(255) NOT NULL,
+    deleted_at TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
+    PRIMARY KEY (namespace)
+);
+
+CREATE UNIQUE INDEX idx_deleted_instances_namespace ON deleted_instances(namespace);
```

Comment:
Is this writing to the queue instance or the control-plane instance?

If this is the queue instance, then we should make sure that the queue instance is dedicated to the one cluster. Up to this point the control-plane's postgres instance has been the sole location for any state related to a data-plane.

If this is the control-plane's DB, then note that this is the first introduction of the pattern of data-plane writing directly back to the control-plane database.

cc @sjmiller609 , @ianstanton

Reply:
We discussed in Slack. It is the queue DB, not the control plane DB, and I think we are both OK with this approach to save a table of deleted instances in the queue db.